src/datatype/network.rs

```diff
@@ -53,7 +53,14 @@ pub struct Url(pub url::Url);
 impl Url {
     /// Append the string suffix to this URL.
     pub fn join(&self, suffix: &str) -> Url {
-        Url(self.0.join(suffix).unwrap_or_else(|err| {
+        let mut url = self.0.clone();
+
+        // add trailing slash if it doesn't exist
+        if url.path_segments().map_or(false, |mut segs| segs.next_back().unwrap() != "") {
```

Comment:
Not easily, no :) The problem is that `path_segments` and `path_segments_mut` return different types, and the mut version has no `peek` method so a pop would be destructive and need to be re-added.